src/backend/cpu/Array.hpp

```diff
@@ -46,22 +46,24 @@ void evalMultiple(std::vector<Array<T> *> arrays);
 
 // Creates a new Array object on the heap and returns a reference to it.
 template<typename T>
-Array<T> createNodeArray(const af::dim4 &size, jit::Node_ptr node);
+Array<T> createNodeArray(const af::dim4 &dims, jit::Node_ptr node);
 
-// Creates a new Array object on the heap and returns a reference to it.
 template<typename T>
-Array<T> createValueArray(const af::dim4 &size, const T &value);
+Array<T> createValueArray(const af::dim4 &dims, const T &value);
 
-// Creates a new Array object on the heap and returns a reference to it.
+// Creates an array and copies from the \p data pointer located in host memory
+//
+// \param[in] dims The dimension of the array
+// \param[in] data The data that will be copied to the array
 template<typename T>
-Array<T> createHostDataArray(const af::dim4 &size, const T *const data);
+Array<T> createHostDataArray(const af::dim4 &dims, const T *const data);
 
 template<typename T>
-Array<T> createDeviceDataArray(const af::dim4 &size, const void *data);
```

Comment:
why is the `const` gone ?

Reply:
The DeviceDataArray uses the pointer from the data array. It may be modified in the future  by another operation so this is not a const parameter.